doc/decisions/base_name.md

```diff
@@ -0,0 +1,33 @@
+# Base Name
+
+## Problem
+
+Both applications and configuration file formats might have arbitrary strings which need to be
+encoded within a key name part.
+
+## Constraints
+
+- keySetBaseName (key, keyBaseName (key)) should be a NOP
+
+## Assumptions
+
+## Considered Alternatives
+
+- restrict what `keySetBaseName` can accept is against:
+  - easy to use of API: applications with some key names would suddenly fail
+  - hard to use it wrong API: storage plugins that do not pre-process key names properly would fail to build up a KeySet
+    which would be against our goal to support any configuration file format (e.g. TOML supports empty key part names).
```

Comment:
> The other one will always have to do pre-/post-processing because the existing format probably has different limitations than Elektra.

Not necessarily: we have plugins which map 1:1, i.e., whatever they parse they pass to `keySetBaseName` or write what they get by `keyGetBaseName`.

> I think this is the correct order

This is exactly the order as written in doc/GOALS.md